cpp/src/arrow/record_batch.h

```diff
@@ -114,6 +114,10 @@ class ARROW_EXPORT RecordBatch {
   /// \return the record batch's schema
   const std::shared_ptr<Schema>& schema() const { return schema_; }
 
+  /// \brief Rewrite schema When the field type is consistent
+  Result<std::shared_ptr<RecordBatch>> RewriteSchema(
```

Comment:
Call it `ReplaceSchema` by analogy with `ReplaceSchemaMetadata`?

Reply:
> Call it `ReplaceSchema` by analogy with `ReplaceSchemaMetadata`?

sure, i will change it.